compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -231,7 +231,9 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
           //     be duplicated
           //  2. To enable correct pickling (calls can share symbols with the inlined code, which
           //     would trigger an assertion when pickling).
-          val callTrace = Ident(call.symbol.topLevelClass.typeRef).withPos(call.pos)
+          val callTrace =
+            if (call.symbol.is(Macro)) call
```

Comment:
Why make the exception for macros? Since there is a reasoning in the comment above why we rewrite calls to callTraces, we need to add to the comment a reasoning why the same does not apply to macro calls.

Reply:
I will add a comment

Reply:
Added the comment and making this optimization at `ReifyQuotes` when the `call` is processed.